src/middlewared/middlewared/plugins/datastore/read.py

```diff
@@ -0,0 +1,267 @@
+from collections import defaultdict
+import re
+
+from sqlalchemy import and_, func, select
+from sqlalchemy.sql import Alias
+
+from middlewared.schema import accepts, Bool, Dict, Int, List, Ref, Str
+from middlewared.service import Service
+from middlewared.service_exception import MatchNotFound
+
+from .filter import FilterMixin
+from .schema import SchemaMixin
+
+
+def regexp(expr, item):
+    reg = re.compile(expr, re.I)
+    return reg.search(item) is not None
+
+
+class DatastoreService(Service, FilterMixin, SchemaMixin):
```

Comment:
FilterMixin already inherits SchemaMixin, not sure you need it here as well?

Reply:
I was thinking that explicit is better than implicit and as I use SchemaMixin methods in this class, I want to state it explicitly.